### DISCUSSION TITLE
Combobox.Target vs Combobox.EventsTarget vs Combobox.DropdownTarget

### DISCUSSION THREAD

**Discussion by jacobwilson41** [mantinedev/mantine#6050] [Q&A] [answered]
## [Combobox.Target](https://mantine.dev/core/combobox/#comboboxtarget)

> `Combobox.Target` should be used as a wrapper for the target element or component. `Combobox.Target` marks its child as a target for dropdown and sets aria-* attributes and adds keyboard event listeners to it.

Questions:
1. **What does "target" mean? Is "target" supposed to be what opens the dropdown?**
2. It appears that `Dropdown.Target`'s child is responsible for handling open/close state with onClick/onFocus/onBlur events.  **So is the only purpose of `Dropdown.Target` to set aria-\* attributes and add keyboard listeners?**
<br/>

## [Combobox.EventsTarget and Combobox.DropdownTarget](https://mantine.dev/core/combobox/#split-events-and-dropdown-targets)

> In some cases, you might need to use different elements as an events target and as a dropdown. Use `Combobox.EventsTarget` to add aria-* attributes and keyboard event handlers, and `Combobox.DropdownTarget` to position the dropdown relative to the target.

Questions:
1. The example provided uses `<Combobox.EventsTarget />` _inside of_ `<Combobox.DropdownTarget />`.   **Are these two components required to be used together?  Is it required that `<Combobox.DropdownTarget />` be a child of `<Combobox.EventsTarget />`?**
2. It appears that `Combobox.Target` and `Combobox.EventsTarget` have the exact same [props](https://mantine.dev/core/combobox/?t=props).  **Are these two components perfectly interchangeable in every way?  If not, what are the differences?**

**Comment by rtivital** [comment, marked as answer]
DropdownTarget is used to calculate dropdown position
EventsTarget is used to add events listeners
Target combines both targets
